pkg/R/sparkR.R

```diff
@@ -149,11 +149,15 @@ sparkR.init <- function(
     }
     f <- file(path, open='rb')
     sparkRBackendPort <- readInt(f)
+    monitorPort <- readInt(f)
     close(f)
     file.remove(path)
-    if (length(sparkRBackendPort) == 0) {
+    if (length(sparkRBackendPort) == 0 || sparkRBackendPort == 0 ||
+     length(monitorPort) == 0 || monitorPort == 0) {
       stop("JVM failed to launch")
     }
+    # never close this socket, JVM is waiting for it
+    .sparkREnv$monitorF <- socketConnection(port = monitorPort)
```

Comment:
Should we clear this variable during `sparkR.stop` ? Also we have a bunch of logic in `sparkR.stop` now that tries to reconnect to the backend if a valid connection does not exist. Can we simplify that logic based on whether this connection is open ?